node-hub/opencv-plot/main.py

```diff
@@ -0,0 +1,179 @@
+
+    args = parser.parse_args()
+
+    plot_width = os.getenv("PLOT_WIDTH", args.plot_width)
+    plot_height = os.getenv("PLOT_HEIGHT", args.plot_height)
+
+    if plot_width is not None:
+        if isinstance(plot_width, str) and plot_width.isnumeric():
+            plot_width = int(plot_width)
+
+    if plot_height is not None:
+        if isinstance(plot_height, str) and plot_height.isnumeric():
+            plot_height = int(plot_height)
+
+    # check if the code is running in a CI environment (e.g. GitHub Actions) (parse to bool)
+    ci_enabled = os.getenv("CI", False)
+    if ci_enabled == "true":
+        ci_enabled = True
+
+    node = Node(args.name)  # provide the name to connect to the dataflow if dynamic node
+    plot = Plot()
+
+    plot.width = plot_width
+    plot.height = plot_height
+
+    pa.array([])  # initialize pyarrow array
+
+    for event in node:
+        event_type = event["type"]
+
+        if event_type == "INPUT":
+            event_id = event["id"]
+
+            if event_id == "tick":
+                node.send_output(
+                    "tick",
+                    pa.array([]),
+                    event["metadata"]
+                )
+
+            elif event_id == "image":
+                arrow_image = event["value"][0]
+                image = {
+                    "width": np.uint32(arrow_image["width"].as_py()),
+                    "height": np.uint32(arrow_image["height"].as_py()),
+                    "channels": np.uint8(arrow_image["channels"].as_py()),
+                    "data": arrow_image["data"].values.to_numpy().astype(np.uint8)
+                }
+
+                plot.frame = np.reshape(image["data"], (image["height"], image["width"], image["channels"]))
+
+                if plot_frame(plot, ci_enabled):
+                    break
+
+            elif event_id == "bbox":
+                arrow_bbox = event["value"][0]
+                plot.bboxes = {
+                    "bbox": arrow_bbox["bbox"].values.to_numpy().reshape(-1, 4),
+                    "conf": arrow_bbox["conf"].values.to_numpy(),
+                    "names": arrow_bbox["names"].values.to_pylist(),
+                }
+
+                if plot_frame(plot, ci_enabled):
+                    break
+
+            elif event_id == "text":
+                arrow_text = event["value"][0]
+                plot.text = {
+                    "text": arrow_text["text"].as_py(),
+                    "font_scale": np.float32(arrow_text["font_scale"].as_py()),
+                    "color": (np.uint8(arrow_text["color"].as_py()[0]),
+                              np.uint8(arrow_text["color"].as_py()[1]),
+                              np.uint8(arrow_text["color"].as_py()[2])),
+                    "thickness": np.uint32(arrow_text["thickness"].as_py()),
+                    "position": (np.uint32(arrow_text["position"].as_py()[0]),
+                                 np.uint32(arrow_text["position"].as_py()[1]))
+                }
+
```

Comment:
So I think that text is commonly associated with a string value. 

Could it be possible to make the text agnostic to the plot, and just have default value for position color.